subsys/fs/Kconfig.fatfs

```diff
@@ -88,6 +89,39 @@ config FS_FATFS_NUM_DIRS
 	  at compile-time.
 	  This affects use of fs_opendir on FAT type mounted file systems.
 
+config FS_FATFS_HAS_RTC
+	bool "Timestamping support"
+	help
+	  Enable file system timestamping instead of using a hardcoded date
+	  for all operations. Requires an application supplied implementation
+	  of `get_fattime`. Format of the uint32_t bits are as follows:
+	    31:25 = Year from 1980
+	    24:21 = Month (1..12)
+	    20:16 = Day of month (1..31)
+	    15:11 = Hour (0..23)
+	    10: 5 = Minute (0..59)
+	     4: 0 = Seconds/2 (0..29)
+
+config FS_FATFS_USE_LABEL
+	bool "Set/Get filesystem labels"
+	help
+	  Enable retrieving and setting the filesystem labels through
+	  `f_getlabel` and `f_setlabel`.
+
+config FS_FATFS_USE_EXPAND
+	bool "Pre-allocate contiguous files"
+	depends on !FS_FATFS_READ_ONLY
+	help
+	  Enable pre-allocating a contiguous data area to the file through
+	  `f_expand`. Unlike expansion through `f_truncate`, `f_expand` does
+	  not write to the expanded memory area.
+
+config FS_FATFS_USE_FIND
+	bool "Native directory iteration"
+	help
+	  Enable the native dictionary iteration functions `f_findfirst` and
+	  `f_findnext`.
+
```

Comment:
Can we have these under "Direct ELM FAT interaction API" or something like that, to separate this from other things that are accessed via the VFS API?

Reply:
The added options already live under a dedicated menu:
https://github.com/zephyrproject-rtos/zephyr/blob/93c9da66944c7c7f79d5c554ca9a640b422d39b3/subsys/fs/Kconfig.fatfs#L15-L16

Reply:
Sorry, I have failed to notice
![image](https://github.com/user-attachments/assets/0451db29-bb1f-4de2-b27f-217caf7d1841)

The reason I was asking for sub-menu is that the options do not regulate how Zephyr interacts with ELM via VFS, or rather user do so, but are rather options enabling direct interaction; actually the F_USE_LABEL* has a small impact on `dir_read`, which is not clearly described in ELM driver.
Since the functions do not impact* the other parts of ELM and additional functions will only be linked when user actually calls them, then we can just have single Kconfig enabling them all. No need to multiply these.

Reply:
You touched on one option that has side effects, but `FF_HAS_RTC` requires the user to provide a function implementation, how would this be handled under a single Kconfig?

Reply:
Yes, two options: one adds the RTC, the other all the extra stuff that is not affecting the VFS usage of ELM driver.